std/algorithm/searching.d

```diff
@@ -1775,7 +1775,11 @@ if (isRandomAccessRange!R1 && isBidirectionalRange!R2
         && is(typeof(binaryFun!pred(haystack.front, needle.front)) : bool))
 {
     if (needle.empty) return haystack;
-    const needleLength = walkLength(needle.save);
+    static if (hasLength!R2) {
```

Comment:
Please follow [the style guide](https://dlang.org/dstyle.html) and have braces on their own lines. Same throughout.